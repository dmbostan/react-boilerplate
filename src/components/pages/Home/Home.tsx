import { useState } from 'react';
import './Home.css';
import { Logos } from '@molecules/index';
import { Paragraph, Heading, Button } from '@atoms/index.ts';

const Home = () => {
  const [count, setCount] = useState(0);

  return (
    <>
      <Logos />
      <Heading as="h1">Vite + React</Heading>
      <div className="card">
        <Button clickHandler={() => setCount((count) => count + 1)}>count is {count}</Button>
        <Paragraph>
          Edit <code>src/App.tsx</code> and save to test HMR
        </Paragraph>
      </div>
      <Paragraph className="read-the-docs">Click on the Vite and React logos to learn more</Paragraph>
    </>
  );
};

export default Home;
