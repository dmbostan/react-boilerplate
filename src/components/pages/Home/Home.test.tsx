import Home from "./Home.tsx";
import { render, screen, fireEvent } from '@testing-library/react';

describe('Home component', async () => {
    it('should have a heading', async () => {
        render(<Home />);
        const h1 = await screen.queryByText('Vite + React');

        expect(h1).toBeInTheDocument();
    });

    it('should show the button count set to 3', async () => {
        await render(<Home />);
        const button = await screen.queryByText('count is 0');

        expect(button).not.toBeNull();

        fireEvent.click(button as HTMLElement);
        fireEvent.click(button as HTMLElement);
        fireEvent.click(button as HTMLElement);

        expect(button?.innerHTML).toBe('count is 3');
    });

    it('should match snapshot', () => {
        const { asFragment} = render(<Home />)

        expect(asFragment()).toMatchSnapshot()
    })
});
