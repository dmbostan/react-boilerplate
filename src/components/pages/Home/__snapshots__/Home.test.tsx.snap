// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > should match snapshot 1`] = `
<DocumentFragment>
  <a
    href="https://vitejs.dev"
    target="_blank"
  >
    <img
      alt="Vite logo"
      class="logo"
      src="/src/assets/vite.svg"
    />
  </a>
  <a
    href="https://react.dev"
    target="_blank"
  >
    <img
      alt="React logo"
      class="logo react"
      src="/src/assets/react.svg"
    />
  </a>
  <h1>
    Vite + React
  </h1>
  <div
    class="card"
  >
    <button>
      count is 0
    </button>
    <p>
      Edit 
      <code>
        src/App.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and React logos to learn more
  </p>
</DocumentFragment>
`;
